src/mc.rs

```diff
@@ -136,72 +140,288 @@ const SUBPEL_FILTERS: [[[i32; SUBPEL_FILTER_SIZE]; 16]; 6] = [
   ]
 ];
 
+// gets an index that can be mapped to a function for a pair of filter modes
+const fn get_2d_mode_idx(mode_x: FilterMode, mode_y: FilterMode) -> usize {
+  mode_x as usize + 4 * (mode_y as usize)
+}
+
+type PutFn =
+  unsafe extern fn(*mut u8, isize, *const u8, isize, i32, i32, i32, i32);
```

Comment:
It would be nice here and for the other `fn`s to give names to the parameters :)